0.3.0 introduced an undocumented dependency on java8 - String.

0.3.0 specifically commit: 2ef687dbda386590f8fe56d7294da6fa14c91628
introduced a dependency on java8 String

I was using the lib inside an android application (mainly running on android6 devices).
project is setup:
- with Java source/target compatibility 8
- kotlin jvm target 8

Project compile without any issue or warning but the app ultimately crash when invoking the codePoints method.

`                 E  java.lang.NoSuchMethodError: No virtual method codePoints()Ljava/util/stream/IntStream; in class Ljava/lang/String;
                    or its super classes (declaration of 'java.lang.String' appears in /system/framework/core-libart.jar)
                 E      at com.jakewharton.picnic.SimpleLayout.draw(textLayout.kt:62)
                 E      at com.jakewharton.picnic.TextRendering.render(textRender.kt:314)
                 E      at com.jakewharton.picnic.TextRendering.render$default(textRender.kt:16)`

--- a/build.gradle
+++ b/build.gradle
@@ -1,3 +1,5 @@
+import org.gradle.internal.jvm.Jvm
+
 buildscript {
   ext.versions = [
       'kotlin': '1.3.60'
@@ -13,6 +15,7 @@ buildscript {
     classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:${versions.kotlin}"
     classpath 'org.jlleitschuh.gradle:ktlint-gradle:9.1.0'
     classpath 'org.jetbrains.dokka:dokka-gradle-plugin:0.10.0'
+    classpath 'ru.vyarus:gradle-animalsniffer-plugin:1.5.0'
   }
 
   repositories {
@@ -46,6 +49,20 @@ subprojects {
         freeCompilerArgs = [
             "-progressive",
         ]
+      }
+    }
+  }
+
+  plugins.withId('java-library') {
+    // Animal Sniffer only works on JDK 11 or older currently.
+    if (!Jvm.current().javaVersion.isJava12Compatible()) {
+      project.apply plugin: 'ru.vyarus.animalsniffer'
+      animalsniffer {
+        sourceSets = [sourceSets.main] // Only check main sources, ignore test code.
+      }
+      dependencies {
+        signature 'org.codehaus.mojo.signature:java18:1.0@signature'
+        signature 'net.sf.androidscents.signature:android-api-level-21:5.0.1_r2@signature'
       }
     }
   }

--- a/picnic/src/main/java/com/jakewharton/picnic/textLayout.kt
+++ b/picnic/src/main/java/com/jakewharton/picnic/textLayout.kt
@@ -59,7 +59,7 @@ internal class SimpleLayout(private val cell: PositionedCell) : TextLayout {
 
     var x = left
     var y = top
-    for (codePoint in cell.cell.content.codePoints()) {
+    for (codePoint in cell.cell.content.codePoints) {
       // TODO invisible chars, graphemes, etc.
       if (codePoint != '\n'.toInt()) {
         canvas[y, x++] = codePoint
@@ -68,5 +68,14 @@ internal class SimpleLayout(private val cell: PositionedCell) : TextLayout {
         x = left
       }
     }
+  }
+}
+
+private val String.codePoints get() = sequence {
+  var i = 0
+  while (i < length) {
+    val codePoint = codePointAt(i)
+    yield(codePoint)
+    i += Character.charCount(codePoint)
   }
 }

